### PR TITLE
Duplicate /Events Endpoint in prison API from custody api

### PR DESCRIPTION
**Pull request by Mjwillis**

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/api/model/OffenderEvent.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+import lombok.NoArgsConstructor;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class OffenderEvent {
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/hmpps/prison/api/model/OffenderEvent.java:15]: Eventually we should break this down into separate classes for each event type -- future work ...

File: src/main/java/uk/gov/justice/hmpps/prison/service/OffenderEventsService.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    public OffenderEventsService(final OffenderEventsTransformer offenderEventsTransformer,
+                                 final OffenderEventsRepository offenderEventsRepository,
+                                 final XtagEventsService xtagEventsService) {
+        this.offenderEventsTransformer = offenderEventsTransformer;
+        this.offenderEventsRepository = offenderEventsRepository;
+        this.xtagEventsService = xtagEventsService;
+    }
+
+    @PreAuthorize("hasRole('PRISON_OFFENDER_EVENTS')")
+    public Optional<List<OffenderEvent>> getEvents(final Optional<LocalDateTime> maybeFrom,
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/hmpps/prison/service/OffenderEventsService.java:47]: I bet you had IntelliJ telling you not to use Optionals as arguments ...

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/impl/BookingResourceImpl.java [hunk 3]
```diff
@@ -204,7 +205,7 @@ public OffenderSummary recallOffenderBooking(@Valid final RecallBooking recallBo
 
     @Override
     public Movement getMovementByBookingIdAndSequence(final Long bookingId, final Integer sequenceNumber) {
-        return movementsService.getMovementByBookingIdAndSequence(bookingId, sequenceNumber);
+        return movementsService.getMovementByBookingIdAndSequence(bookingId, sequenceNumber).orElseThrow(EntityNotFoundException.withMessage(format("Movement Not found booking Id %d, seq %d", bookingId, sequenceNumber)));
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/impl/BookingResourceImpl.java:208]: Why did you have to change the service to Optional?

File: src/main/java/uk/gov/justice/hmpps/prison/api/model/OffenderEvent.java (hunk 1 above)
Comment (Mjwillis) [reply src/main/java/uk/gov/justice/hmpps/prison/api/model/OffenderEvent.java:15]: Agreed there is lots to improve!

File: src/main/java/uk/gov/justice/hmpps/prison/service/OffenderEventsService.java (hunk 2 above)
Comment (Mjwillis) [reply src/main/java/uk/gov/justice/hmpps/prison/service/OffenderEventsService.java:47]: it is

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/impl/BookingResourceImpl.java (hunk 3 above)
Comment (Mjwillis) [reply src/main/java/uk/gov/justice/hmpps/prison/api/resource/impl/BookingResourceImpl.java:208]: I did it to make it more consistent with custody api code and also realised there would be a null pointer if no movement was found.

**Pull request merged**